tests/system/utils.py

```diff
@@ -39,3 +40,122 @@ def get_resource(resource):
                 raise Exception
         except Exception:
             raise DCOSException("Can't read from resource: {0}. Please check that it exists.".format(resource))
+
+
+class FaultDomain:
+    """
+    High-level representation of a fault domain
+    """
+    def __init__(self, config):
+        # Make sure config is a dict
+        if not type(config) is dict:
+            config = {}
+
+        # Extract fault domain
+        fault_domain = config.get('fault_domain', {})
+
+        # Extract zone/region
+        self.zone = fault_domain.get('zone', {}).get('name', 'default')
+        self.region = fault_domain.get('region', {}).get('name', 'default')
+
+
+def get_cluster_local_domain():
+    """Contacts the DC/OS mesos master and returns it's faultDomain configuration (aka "local domain")
+    """
+    master = mesos.get_master()
+    return FaultDomain(master.state().get('domain', {}))
+
+
+def get_cluster_agent_domains():
+    """Returns a dictionary with the slave IDs in the cluster and their corresponding
+       fault domain information
+    """
+    agent_domains = {}
+
+    # Populate agent_domains with the ID and the corresponding domain for each slave
+    master = mesos.get_master()
+    for slave in master.slaves():
+        agent_domains[slave['id']] = FaultDomain(slave._short_state.get('domain', None))
+    return agent_domains
+
+
+def get_all_cluster_regions():
+    """Returns a dictionary with all the regions and their zones in the cluster
```

Comment:
So it's `region -> list[zone]`?

Reply:
Yes, but in some cases you also want the name of that zone, hence the tuple approach.
Honestly, if you want to make things more verbose you could do:
```
region = get_biggest_region_name()
zones = get_region_zones(region)
```

This is just compacting the two methods in one:
```
(region, zones) = get_biggest_region()
```

Reply:
I am assuming there are no other comments here?